lib/reverse_words.rb

```diff
@@ -1,4 +1,33 @@
 # A method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  raise NotImplementedError
+  if my_words == nil || my_words.length == 0
+    return my_words
+  end
+
+  i = 0
+  j = 0
+  max = 0
+  
+  until max == my_words.length - 1
+    until my_words[i] != " "
+      i += 1
+      j += 1
+    end
+
+    until my_words[j + 1] == " " || j == my_words.length - 1
+      j += 1
+    end
+
+    max = j
+
+    while i < j
+      x = my_words[i]
+      my_words[i] = my_words[j]
+      my_words[j] = x
+      i += 1
+      j -= 1
+    end
+
+    i = max + 1
```

Comment:
Your algorithm does work currently for the test cases in place. At the end of the loop ending on line 29 however, i and j have the same values. Then, you update i to be max+1 on line 31. Which means in the most typical scenarios, j will be less than i after line 31. For efficiency, it might be worthwhile to make j also the same value.